clients/buckets/bucket.go

```diff
@@ -369,7 +381,7 @@ func (c Client) upsert(ctx context.Context, bucketName string, data []byte) (Res
 	if b, err := unmarshalJSON(apiErr.Body); err != nil && b.Status == "deleting" {
 		logger.V(1).Info(fmt.Sprintf("Bucket '%s' is being deleted. Waiting before re-creation...", b.BucketName))
 		if err := c.awaitBucketRemoved(ctx, bucketName); err != nil {
-			return Response{}, err
+			return api.Response{}, err
 		}
 		return c.Create(ctx, bucketName, data)
 	}
```

Comment:
Is this dead code? First, it should be `err == nil`, and second, the response doesn't contain the bucket status.
This is the response I get while the bucket is being deleted.
![image](https://github.com/user-attachments/assets/cc3658c6-fccb-4673-9a1f-e7612daaeaf0)